test/test_fully_connected_layer.h

```diff
@@ -10,9 +10,51 @@
 #include "testhelper.h"
 #include "tiny_dnn/tiny_dnn.h"
 
+namespace tiny_dnn {
+
 using namespace tiny_dnn::activation;
 
-namespace tiny_dnn {
+TEST(fully_connected, setup_internal) {
+  fully_connected_layer l(32, 10, true, core::backend_t::internal);
+
+  EXPECT_EQ(l.parallelize(), true);            // if layer can be parallelized
+  EXPECT_EQ(l.in_channels(), 1u);              // num of input tensors
+  EXPECT_EQ(l.out_channels(), 1u);             // num of output tensors
+  EXPECT_EQ(l.in_data_size(), 32u);            // size of input tensors
+  EXPECT_EQ(l.out_data_size(), 10u);           // size of output tensors
+  EXPECT_EQ(l.fan_in_size(), 32u);             // num of incoming connections
+  EXPECT_EQ(l.fan_out_size(), 10u);            // num of outgoing connections
+  EXPECT_EQ(l.parameters().size(), 2u);        // num of trainable parameters
+  EXPECT_EQ(l.ith_parameter(0).size(), 320u);  // size of weight parameter
+  EXPECT_EQ(l.ith_parameter(1).size(), 10u);   // size of bias parameter
+  EXPECT_STREQ(l.layer_type().c_str(),
+               "fully-connected");  // string with layer type
+}
+
+TEST(fully_connected, forward) {
+  fully_connected_layer l(3, 2, true);
+
+  vec_t in = {1, 2, 3};
+  // clang-format off
+  vec_t weights = {
+    2, 6, 4,
+    4, 6, 2
+  };
+  vec_t bias = {-4, 22};
+  // clang-format on
+  vec_t expected = {24, 42};
+
+  l.ith_parameter(0).set_data(Tensor<float_t>(weights));
+  l.ith_parameter(1).set_data(Tensor<float_t>(bias));
```

Comment:
@karandesai-96 @Randl @prlz77 what do you think about having `set_weights(...)` and `set_bias(...)` per layer?
 it will be more informative for the user